common/src/main/java/fudge/notenoughcrashes/ModConfig.java

```diff
@@ -3,26 +3,86 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import fudge.notenoughcrashes.platform.NecPlatform;
-import net.fabricmc.loader.api.FabricLoader;
 
 import java.io.*;
 
 public class ModConfig {
 
     public enum CrashLogUploadType {
-        GIST
+        GIST(5), // attempt last
+        HASTE(2),
+        PASTEBIN(), // requires configuration
+        BYTEBIN(1);
+
+        private final int defaultPriority;
+
+        CrashLogUploadType(int defaultPriority) { this.defaultPriority = defaultPriority; }
+        CrashLogUploadType() { this.defaultPriority = Integer.MIN_VALUE; }
+        public int getPriority() { return this.defaultPriority; }
     }
 
+    public enum PastebinPrivacy {
+        PUBLIC("0"), //anyone can see it, appears in recently created
+        UNLISTED("1"); // only people with the link can see it
+//        PRIVATE(2) // only you can see it (doesn't make much sense). this doesn't allow for raw download, so disabling
+
+        private final String apiValue;
+
+        PastebinPrivacy(String apiValue) {
+            this.apiValue = apiValue;
+        }
+
+        public String getApiValue() {
+            return this.apiValue;
+        }
+    }
+
+    public enum PastebinExpiry {
+        NEVER("N"),
+        TENMIN("10M"),
+        ONEHOUR("1H"),
+        ONEDAY("1D"),
+        ONEWEEK("1W"),
+        TWOWEEK("2W"),
+        ONEMONTH("1M"),
+        SIXMONTH("6M"),
+        ONEYEAR("1Y");
+
+        private final String pastebinExpiryKey;
+        PastebinExpiry(String pastebinExpiry) {
+            this.pastebinExpiryKey = pastebinExpiry;
+        }
+
+        public String getPastebinExpiryKey() {
+            return pastebinExpiryKey;
+        }
+    }
     private static final File CONFIG_FILE = new File(NecPlatform.instance().getConfigDirectory().toFile(), NotEnoughCrashes.MOD_ID + ".json");
     private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
     private static ModConfig instance = null;
 
+
+
     public CrashLogUploadType uploadCrashLogTo = CrashLogUploadType.GIST;
+    public String uploadCustomUserAgent = null;
+
+    public String GISTUploadKey = "";
```

Comment:
Use null instead of empty string as the default value.

Reply:
null hides it from the config file